iroha/src/crypto.rs

```diff
@@ -106,6 +107,66 @@ impl Debug for Signature {
     }
 }
 
+/// Describes an object that can be signed.
+pub trait Sign {
+    /// The bytes that are signed.
+    fn payload(&self) -> Vec<u8>;
+
+    /// Method to get the signature based on the `payload`.
+    fn signature(
+        &self,
+        public_key: &PublicKey,
+        private_key: &PrivateKey,
+    ) -> Result<Signature, String> {
+        Ok(Signature::new(*public_key, &self.payload(), private_key)?)
+    }
+}
+
+/// Container for multiple signatures.
+#[derive(Debug, Clone, Encode, Decode, Default)]
+pub struct Signatures {
+    signatures: BTreeMap<PublicKey, Signature>,
+}
+
+impl Signatures {
+    /// Adds multiple signatures and skips the duplicates.
+    pub fn add_multiple(&mut self, signatures: &[Signature]) {
+        for signature in signatures.iter().cloned() {
+            let _result = self.try_add(signature);
```

Comment:
`eprintln` in case of errors to do not miss them.

Reply:
It is not actually an error, it basically means that the signature was not added as it was already there. So I am not sure if we should print it.

Reply:
o, okay